.travis.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# CI testing script
+# This is meant to be called from .travis.yml -- it works around limitations
+
+set -e # Error build immediately if install script exits with non-zero
+
+# if this is a docker job, set up and then recurse into the container,
+# instead of continuing.
+# TODO: figure out and pass needed environment variables individually with -e
+
+if ! [ -z "$IMAGE" ]; then
```

Comment:
For a bit more clarity maybe use better names than IMAGE and DEPS?
eg. CONTAINER_IMAGE / CONTAINER_INSTALL_DEPS_CMDLINE

Also `! [ -z ]` could be `[ -n ]` but that's nothing.

Reply:
Yes and yes. I like both of those. I'll fix them up.

Reply:
Yep, I'll do this!